lib/matplotlib/artist.py

```diff
@@ -1593,52 +1597,59 @@ def getp(obj, property=None):
 
 def setp(obj, *args, file=None, **kwargs):
     """
-    Set a property on an artist object.
+    Set one or more properties on an `.Artist`, or list allowed values.
 
-    matplotlib supports the use of :func:`setp` ("set property") and
-    :func:`getp` to set and get object properties, as well as to do
-    introspection on the object.  For example, to set the linestyle of a
-    line to be dashed, you can do::
+    Parameters
+    ----------
+    obj : `.Artist` or list of `.Artist`
+        The artist(s) whose properties are being set or queried.  When setting
+        properties, all artists are affected; when querying the allowed values,
+        only the first instance in the sequence is queried.
 
-      >>> line, = plot([1, 2, 3])
-      >>> setp(line, linestyle='--')
+        For example, two lines can be made thicker and red with a single call:
 
-    If you want to know the valid types of arguments, you can provide
-    the name of the property you want to set without a value::
+        >>> x = arange(0, 1, 0.01)
+        >>> lines = plot(x, sin(2*pi*x), x, sin(4*pi*x))
+        >>> setp(lines, linewidth=2, color='r')
 
-      >>> setp(line, 'linestyle')
-          linestyle: {'-', '--', '-.', ':', '', (offset, on-off-seq), ...}
+    file : file-like or None, default: `sys.stdout`
+        Where `setp` writes its output, e.g. ::
```

Comment:
This defaults description only makes sense if you *know* we're forwarding this to `print`, otherwise `None` could also plausibly interpreted to mean "no output".

```suggestion
    file : file-like, optional, default: `sys.stdout`
        Where `setp` writes its output if it is asked to list allowed values.
        e.g. ::

            with open('output.log') as file:
                setp(line, file=file)

        Passing ``None`` forwards output to `sys.stdout`. 
```

Reply:
yup